### PR TITLE
Adding UvA-DL notebook "Self-Supervised Contrastive Learning with SimCLR"

### PR DESCRIPTION
**Pull request by phlippe**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
   => Discussed on Slack
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

This PR adds a new tutorial of the UvA Deep Learning course, this time on Contrastive Learning with SimCLR. The tutorial is an adapted version of the original tutorial for the Deep Learning course at the University of Amsterdam ([compiled version](https://uvadlc-notebooks.readthedocs.io/en/latest/tutorial_notebooks/tutorial17/SimCLR.html)).

The png, jpg, and svg files are images that are included in the notebook.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananyahjha93 — APPROVED** [review]

**Comment by ananyahjha93** [comment]
LGTM for an intro to contrastive learning tutorial!

File: course_UvA-DL/contrastive-learning/SimCLR.py [hunk 1]
```diff
@@ -0,0 +1,865 @@
...
+# Further, self-supervised learning can provide an alternative to transfer learning from models pretrained on ImageNet since we could pretrain a model on a specific dataset/situation, e.g. traffic scenarios for autonomous driving.
+#
+# Within the last two years, a lot of new approaches have been proposed for self-supervised learning, in particular for images, that have resulted in great improvements over supervised models when few labels are available.
+# The subfield that we will focus on in this tutorial is contrastive learning.
+# Contrastive learning is motivated by the question mentioned above: how are images different from each other?
+# Specifically, contrastive learning methods train a model to cluster an image and its slightly augmented version in latent space, while the distance to other images should be maximized.
+# A very recent and simple method for this is [SimCLR](https://arxiv.org/abs/2006.10029), which is visualized below (figure credit - [Ting Chen et al. ](https://simclr.github.io/)).
+#
+# <center width="100%"><img src="simclr_contrastive_learning.png" width="500px"></center>
```
Comment (Borda) [inline course_UvA-DL/contrastive-learning/SimCLR.py:15]: as MD?

File: course_UvA-DL/contrastive-learning/SimCLR.py [hunk 2]
```diff
@@ -0,0 +1,865 @@
...
+        )
+
+    def configure_optimizers(self):
+        optimizer = optim.AdamW(self.parameters(), lr=self.hparams.lr, weight_decay=self.hparams.weight_decay)
+        lr_scheduler = optim.lr_scheduler.CosineAnnealingLR(
+            optimizer, T_max=self.hparams.max_epochs, eta_min=self.hparams.lr / 50
+        )
+        return [optimizer], [lr_scheduler]
+
+    def info_nce_loss(self, batch, mode='train'):
```
Comment (Borda) [inline course_UvA-DL/contrastive-learning/SimCLR.py:299]: rather full name instead of `nce`

File: course_UvA-DL/contrastive-learning/SimCLR.py [hunk 3]
```diff
@@ -0,0 +1,865 @@
...
+        num_workers=0
+    )
+    test_loader = data.DataLoader(
+        test_feats_data,
+        batch_size=batch_size,
+        shuffle=False,
+        drop_last=False,
+        pin_memory=True,
+        num_workers=0
+    )
```
Comment (Borda) [inline course_UvA-DL/contrastive-learning/SimCLR.py:576]: shall we wrap it into dataModule so then do not need to repeat the datalodder definition always...

Comment (Borda) [reply course_UvA-DL/contrastive-learning/SimCLR.py:576]: also the reader is ensured that the training data are identical and be wondering if some argument changed...

**Review by Borda — APPROVED** [review]
let's insert all the PNG/JPG images as MD

File: course_UvA-DL/contrastive-learning/SimCLR.py (hunk 2 above)
Comment (phlippe) [reply course_UvA-DL/contrastive-learning/SimCLR.py]: Is `info_noise_contrastive_estimation_loss` more informative than `info_nce_loss` though? I think InfoNCE is more popular than the long version, but I am also ok with the long name.

File: course_UvA-DL/contrastive-learning/SimCLR.py (hunk 1 above)
Comment (phlippe) [reply course_UvA-DL/contrastive-learning/SimCLR.py]: I have added the png images in the following format now:
`<center width="100%"> simclr contrastive learning{width="500px"} </center>`
It compiled correctly locally and seems to pass the documentation test as well.

File: course_UvA-DL/contrastive-learning/SimCLR.py (hunk 2 above)
Comment (Borda) [reply course_UvA-DL/contrastive-learning/SimCLR.py]: I would say so, but just my opinions (Python is talkative languge)

File: course_UvA-DL/contrastive-learning/SimCLR.py (hunk 1 above)
Comment (Borda) [reply course_UvA-DL/contrastive-learning/SimCLR.py]: cool!

**Review by tchaton — APPROVED** [review]

**Pull request merged**